lib/ansible/modules/storage/netapp/na_elementsw_snapshot_restore.py

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/python
+
+# (c) 2018, NetApp, Inc
+# GNU General Public License v3.0+ (see COPYING or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+
+"""
+Element Software Snapshot Restore
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+
+module: na_elementsw_snapshot_restore
+
+short_description:  NetApp Element Software Restore Snapshot
+extends_documentation_fragment:
+    - netapp.solidfire
+version_added: '2.7'
+author: NetApp Ansible Team (ng-ansibleteam@netapp.com)
+description:
+- Element OS Cluster restore snapshot to volume.
+
+options:
+
+    src_volume_id:
+        description:
+        - ID or Name of source active volume.
+        required: true
+
+    src_snapshot_id:
+        description:
+        - ID or Name of an existing snapshot.
+        required: true
+
+    dest_volume_name:
+        description:
+        - New Name of destination for restoring the snapshot
+        required: true
+
+    account_id:
+        description:
+        - Account ID or Name of Parent/Source Volume.
+        required: true
+
+    password:
```

Comment:
Why are username and password define here, though should be in the docs_fragment.